forward/utils.go

```diff
@@ -17,9 +17,27 @@ func cloneURL(i *url.URL) *url.URL {
 // copyHeaders copies http headers from source to destination.  It does not
 // overide, but adds multiple headers
 func copyHeaders(dst, src http.Header) {
+	var extraHopByHopHeaders []string
 	for k, vv := range src {
-		for _, v := range vv {
-			dst.Add(k, v)
+		switch k {
```

Comment:
An array lookup is basically the same as a switch, it is O(n). At least in an array lookup you know for sure that it will benefit from processor cache (although probably the switch-case case too).
The one that might be more efficient is a map, but probably only for larger sets. I personally find it more natural to express it in terms of a map, but that's just personal opinion.
This is how Oxy is implementing this, which I didn't add in my rewrite of forward: https://github.com/mailgun/oxy/blob/547c334d658398c05b346c0b79d8f47ba2e1473b/forward/rewrite.go#L47